[BUG] Linter dies with panic when empty return is called

**Describe the problem**

When `return;` is called from a subroutine, `stmt.Ident.Value` is `nil`:
https://github.com/ysugimoto/falco/blob/v0.2.2/linter/linter.go#L1012

which can end up in SIGSEGV:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x119d4c3]
```

As the code comment suggests, it is allowed to call the `return` statement without specifying any state:
https://github.com/ysugimoto/falco/blob/v0.2.2/parser/statement_parser.go#L367-L373

and it is only allowed from user-defined custom subroutines. Otherwise an error with:

> Empty returns not accessible in state-machine method vcl_xxxx


**VCL code that cause the problem**

```
sub vcl_recv {
  return;
}
```

or 

```
sub foo {
  return;
}
```

**Expected behavior**

Do not cause panic, and accept empty returns from user-defined custom subroutines. Think the subroutine name that the linter is currently processing needs to be accessible from `lintReturnStatement` function so that the linter could skip error if `stmt.Ident.Value` is `nil` and the subroutine is not a Fastly reserved subroutine.

--- a/linter/linter.go
+++ b/linter/linter.go
@@ -1009,6 +1009,11 @@ func (l *Linter) lintReturnStatement(stmt *ast.ReturnStatement, ctx *context.Con
 		expects = append(expects, "deliver")
 	}
 
+	// return statement may not have arguments, then stop linting
+	if stmt.Ident == nil {
+		return types.NeverType
+	}
+
 	if !expectState(stmt.Ident.Value, expects...) {
 		l.Error(InvalidReturnState(
 			stmt.Ident.GetMeta(), context.ScopeString(ctx.Mode()), stmt.Ident.Value, expects...,

--- a/linter/linter_test.go
+++ b/linter/linter_test.go
@@ -1186,3 +1186,22 @@ sub foo {
 		assertNoError(t, input)
 	})
 }
+
+func TestReturnStatement(t *testing.T) {
+	t.Run("pass: without argument", func(t *testing.T) {
+		input := `
+sub foo {
+	return;
+}`
+		assertNoError(t, input)
+	})
+
+	t.Run("pass: with argument", func(t *testing.T) {
+		input := `
+sub vcl_recv {
+	#Fastly recv
+	return (pass);
+}`
+		assertNoError(t, input)
+	})
+}

